client/cypress/end-to-end/ballot-comparison-failure-cases.spec.js

```diff
@@ -0,0 +1,238 @@
+    cy.findAllByText('Upload File').spread((firstButton, secondButton) => {
+      firstButton.click()
+    })
+    cy.contains("You must upload a file")
+
+    // upload invalid jurisdiction filesheet
+    cy.fixture('CSVs/jurisdiction/sample_jurisdiction_filesheet_jurisdiction_col_error.csv').then(fileContent => {
+      cy.get('input[type="file"]').first().attachFile({
+        fileContent: fileContent.toString(),
+        fileName: 'sample_jurisdiction_filesheet_jurisdiction_col_error.csv',
+        mimeType: 'csv'
+      })
+    })
+    cy.findAllByText('Upload File').spread((firstButton, secondButton) => {
+      firstButton.click()
+    })
+    cy.get('.Toastify').find('div').find('div').contains('Missing required CSV field "Jurisdiction"').invoke('text')
+    .then(text => {
+      const toastText = text
+      expect(toastText).to.equal('Missing required CSV field "Jurisdiction"')
+    })
+
+    // upload valid jurisdiction filesheet
+    cy.findAllByText('Upload File').should('have.length',2)
+    cy.fixture('CSVs/jurisdiction/sample_jurisdiction_filesheet.csv').then(fileContent => {
+      cy.get('input[type="file"]').first().attachFile({
+        fileContent: fileContent.toString(),
+        fileName: 'sample_jurisdiction_filesheet.csv',
+        mimeType: 'csv'
+      })
+    })
+    cy.findAllByText('Upload File').spread((firstButton, secondButton) => {
+      firstButton.click()
+    })
+    cy.contains("Upload successfully completed")
+
+    // attempt uploading a standardized contests file without selecting one
+    cy.findAllByText('Upload File').click()
+    cy.contains("You must upload a file")
+
+    // upload invalid standardized contests file
+    cy.fixture('CSVs/contest/sample_standardized_contests_contest_name_col_error.csv').then(fileContent => {
+      cy.get('input[type="file"]').last().attachFile({
+        fileContent: fileContent.toString(),
+        fileName: 'sample_standardized_contests_contest_name_col_error.csv',
+        mimeType: 'csv'
+      })
+    })
+    cy.findAllByText('Upload File').click()
+    cy.contains("Missing required column: Contest Name.")
+
+    // upload valid standardized contests file
+    cy.findAllByText('Replace File').spread((firstButton, secondButton) => {
+      secondButton.click()
+    })
+    cy.fixture('CSVs/contest/ballot_comparison_contests.csv').then(fileContent => {
+      cy.get('input[type="file"]').last().attachFile({
+        fileContent: fileContent.toString(),
+        fileName: 'ballot_comparison_contests.csv',
+        mimeType: 'csv'
+      })
+    })
+    cy.findAllByText('Upload File').click()
+    cy.findAllByText(/Upload successfully completed/).should('have.length', 2)
+    cy.get('button[type="submit"]').should('not.have.class', 'bp3-disabled').click()
+    cy.findAllByText('Target Contests').should('have.length', 2)
+
+    // neglect to select a targeted contest
```

Comment:
should this even be allowed in the frontend? i.e. should the save & next button just be disabled?

Reply:
Hadn't thought of that, but quite possibly. What are your thoughts @mcchilders ?